### PR TITLE
Support for Linseed access tokens

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Operator changes so that Linseed can provision access tokens, and 
RBAC changes for Linseed clients.

- Create a keypair for Linseed to use when generating access tokens.
- RBAC for Linseed to create tokens in managed clusters.
- Mount tokens into Linseed clients in managed clusters.
- RBAC for Linseed clients

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by asincu** [comment]
@caseydavenport Don;t we also need to need a similar setup for intrusion controller ? It should receive a token to access Linseed and get audit, flows, l7, waf, dns, events and create events. Also DPI should get a token to create events.

File: pkg/render/compliance.go [hunk 1]
```diff
@@ -325,23 +330,28 @@ func (c *complianceComponent) complianceControllerClusterRole() *rbacv1.ClusterR
 				Resources: []string{"globalreports/finalizers"},
 				Verbs:     []string{"update"},
 			},
+			{
+				APIGroups: []string{"linseed.tigera.io"},
+				Resources: []string{"compliancereports"},
+				Verbs:     []string{"create"},
```
Comment (asincu) [inline pkg/render/compliance.go:336]: I do not remember if the controller also needed access to get the latest generated report.

File: pkg/render/intrusion_detection.go [hunk 2]
```diff
@@ -444,6 +456,26 @@ func (c *intrusionDetectionComponent) intrusionDetectionClusterRole() *rbacv1.Cl
...
+		},
+		{
+			// Add read access to Linseed APIs.
+			APIGroups: []string{"linseed.tigera.io"},
+			Resources: []string{
+				"waflogs",
+				"dnslogs",
+				"l7logs",
+				"flows",
+				"kube_auditlogs",
```
Comment (asincu) [inline pkg/render/intrusion_detection.go:473]: I think we also need any audit logs.

File: pkg/render/intrusiondetection/dpi/dpi_test.go [hunk 3]
```diff
@@ -299,6 +304,7 @@ var _ = Describe("DPI rendering tests", func() {
 			{name: dpi.DeepPacketInspectionName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
 			{name: dpi.DeepPacketInspectionName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
 			{name: dpi.DeepPacketInspectionName, ns: dpi.DeepPacketInspectionNamespace, group: "apps", version: "v1", kind: "DaemonSet"},
+			{name: "tigera-linseed", ns: "tigera-intrusion-detection", group: "rbac.authorization.k8s.io", version: "v1", kind: "RoleBinding"},
```
Comment (asincu) [inline pkg/render/intrusiondetection/dpi/dpi_test.go]: Should be `tigera-dpi`.

File: pkg/render/intrusiondetection/dpi/dpi_test.go [hunk 4]
```diff
@@ -321,6 +327,7 @@ var _ = Describe("DPI rendering tests", func() {
...
```
Comment (asincu) [inline pkg/render/intrusiondetection/dpi/dpi_test.go]: Should be `tigera-dpi`.

File: pkg/render/logstorage.go [hunk 5]
```diff
@@ -462,6 +467,48 @@ func (es *elasticsearchComponent) Ready() bool {
...
+	// specific namespaces using RoleBindings so that we only have permissions in our namespaces.
+	role := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "tigera-linseed",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"secrets"},
+				Verbs:     []string{"create", "update", "get", "list"},
```
Comment (asincu) [inline pkg/render/logstorage.go:483]: Will we need delete in the future? We would probably need to think about clean -up when a managed cluster is disconnected.

File: pkg/render/manager.go [hunk 6]
```diff
@@ -674,6 +685,33 @@ func managerClusterRole(managementCluster, managedCluster, usePSP bool) *rbacv1.
...
+			{
+				// Add access to Linseed APIs.
+				APIGroups: []string{"linseed.tigera.io"},
+				Resources: []string{
+					"flows",
+					"flowlogs",
+					"bgplogs",
+					"auditlogs",
+					"kube_auditlogs",
+					"ee_auditlogs",
```
Comment (asincu) [inline pkg/render/manager.go:697]: Do we also need any kube logs ?

Comment (caseydavenport) [reply pkg/render/manager.go:697]: "auditlogs" is any audit logs - it's two lines up.

**Pull request merged**